packages/core/src/node/kill-tree.ts

```diff
@@ -0,0 +1,129 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import * as cp from 'child_process';
+import * as readline from 'readline';
+
+/**
+ * When this script is invoked as a standalone, the first argument must be the pid to kill the tree from.
+ */
+if (require.main === module) {
+    // Shield this process from erroneously exiting on SIGTERM, as this script might try to kill itself:
+    process.on('SIGTERM', () => process.exit(0));
```

Comment:
I'm not sure I get the logic here. If the goal is to avoid 'erroneously exiting on SIGTERM,' is it correct to exit on SIGTERM?

Reply:
It normalizes the exit to `0` for both no errors and receiving its own SIGTERM. Makes handling of the process easier at the caller level.

Reply:
Not needed anymore.